syntax = "proto3";


option java_package = "jarvis.sms";
option java_outer_classname = "JarvisSMSProto";
option objc_class_prefix = "JARVISSMS";

package jarvis.sms;

// ==== AU SMS ====
// au sms request.
message AuSMSRequest {
    string mobile_number = 1; // au photo number
    string message = 2; // message
    string app = 3; // message come from.
}

// au sms request.
message AuAdminSMSRequest {
    string message = 1; // message
    string app = 2; // message come from.
}

// au sms response, as configured by the request.
message AuSMSResponse {
    bool success = 1; // successful.
    string detail = 2; // detail or err msg.
}

// === CN SMS ===
// cn sms request
message CnSMSRequest {
    message PhoneNumber {
        string number = 1;
    }
    repeated PhoneNumber mobile_numbers = 1; // au photo number
    string message = 2; // message
    string app = 3; // message come from.
}

// CN sms response, as configured by the request.
message CnSMSResponse {
    bool success = 1; // successful.
    string detail = 2; // detail or err msg.
}

// === Service ===
service SMS {
    // AU SMS
    rpc SMSAdmin(AuAdminSMSRequest) returns (AuSMSResponse);
    rpc SMSAu(AuSMSRequest) returns (AuSMSResponse);

    // CN SMS
    rpc SMSCn(CnSMSRequest) returns (CnSMSResponse);

}
